src/main/java/me/zeroeightsix/kami/feature/module/player/Blink.java

```diff
@@ -1,47 +1,79 @@
 package me.zeroeightsix.kami.feature.module.player;
 
+import io.github.fablabsmc.fablabs.api.fiber.v1.annotation.Setting;
 import me.zero.alpine.listener.EventHandler;
 import me.zero.alpine.listener.Listener;
 import me.zeroeightsix.kami.event.PacketEvent;
 import me.zeroeightsix.kami.feature.module.Module;
+import me.zeroeightsix.kami.util.Texts;
 import net.minecraft.client.network.OtherClientPlayerEntity;
 import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.network.Packet;
 import net.minecraft.network.packet.c2s.play.PlayerMoveC2SPacket;
+import net.minecraft.util.Formatting;
 
 import java.util.LinkedList;
 import java.util.Queue;
 
+
 @Module.Info(name = "Blink", category = Module.Category.PLAYER)
 public class Blink extends Module {
+    @Setting
+    private boolean packetDiscard = true;
+    @Setting(comment = "Withhold all packets, not just PlayerMoveC2SPacket")
+    private boolean withholdAllPackets = true;
+    @Setting
+    private @Setting.Constrain.Range(min = 50, max = 500, step = Double.MIN_VALUE) int maxPacketAmount = 100;
+
+    private OtherClientPlayerEntity clonedPlayer; // Fake Player when player blinked
+    Queue<Packet> packets = new LinkedList<>(); // Create list to hold our packets
 
-    Queue<PlayerMoveC2SPacket> packets = new LinkedList<>();
     @EventHandler
     public Listener<PacketEvent.Send> listener = new Listener<>(event -> {
-        if (event.getPacket() instanceof PlayerMoveC2SPacket) {
+        // We don't want to withhold login packets if a player logs out with blink enabled
+        if (event.getPacket() instanceof PlayerMoveC2SPacket || (withholdAllPackets && mc.world != null)) {
             event.cancel();
-            packets.add((PlayerMoveC2SPacket) event.getPacket());
+            packets.add(event.getPacket());
+
+            // Create a sense of urgency to unblick when approaching packet threshold
+            Formatting packetColor = Formatting.WHITE;
+            if (packets.size() > maxPacketAmount*(0.6f)) packetColor = Formatting.YELLOW;
+            if (packets.size() > maxPacketAmount) packetColor = Formatting.RED;
+
+            // Shows up above hotbar
+            mc.player.sendMessage(Texts.f(Formatting.WHITE, Texts.append(
+                    Texts.lit("Packets "),
+                    Texts.flit(packetColor, Integer.toString(packets.size())),
+                    Texts.flit(Formatting.WHITE, "/"+(withholdAllPackets?Integer.toString(maxPacketAmount):"∞")))),
+                    true);
         }
```

Comment:
Why the double indent?

Reply:
My IDE was being odd, I didn't catch it.